lib/instance_billing_flavor_check/utils.py

```diff
@@ -0,0 +1,187 @@
+#
+# You should have received a copy of the GNU General Public License along with
+# instance-billing-flavor-check. If not, see <http://www.gnu.org/licenses/>.
+
+import csv
+import configparser
+import ipaddress
+import logging
+import sys
+import requests
+
+from instance_billing_flavor_check.command import Command
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    handlers=[
+        logging.StreamHandler(sys.stdout),
+        logging.FileHandler('instance-billing-flavor-check.log', mode='w'),
+    ],
+    level=logging.INFO,
+    format="%(message)s"
+)
+try:
+    from cloudregister.registerutils import get_smt
+except ModuleNotFoundError as err:
+    pass
+
+REGION_SRV_CLIENT_CONFIG_PATH = '/etc/regionserverclnt.cfg'
+ETC_OS_RELEASE_PATH = '/etc/os-release'
+ETC_HOSTS_PATH = '/etc/hosts'
+
+requests.packages.urllib3.disable_warnings(
+    requests.packages.urllib3.exceptions.InsecureRequestWarning
+)
+
+ 
+def get_instance_data_command():
+    config = configparser.ConfigParser()
+    if config.read(REGION_SRV_CLIENT_CONFIG_PATH):
+        try:
+            return config['instance']['dataProvider']
+        except Exception as err:
+            logger.error(
+                "Could not parse %s: %s", REGION_SRV_CLIENT_CONFIG_PATH, err
+            )
+    else:
+        logger.error("Could not read file %s", REGION_SRV_CLIENT_CONFIG_PATH)
+
+
+def get_metadata():
+    """Return instance metadata."""
+    command = get_instance_data_command()
+    if not command:
+        return
+
+    command = command.split(' ')
+    result = Command.run(command)
+    if result.returncode == 0:
+        return result.output
+
+    logger.error(
+        "Could not fetch the metadata after running the command '%s': '%s' with status '%s'",
+        ' '.join(command),
+        result.error,
+        result.returncode
+    )
+
+
+def get_identifier():
+    """Return the identifier found in /etc/os-release."""
+    try:
+        with open(ETC_OS_RELEASE_PATH, encoding='utf-8') as stream:
+            csv_reader = csv.reader(stream, delimiter="=")
+            os_release = dict(csv_reader)
+        return os_release.get('NAME')
+    except FileNotFoundError:
+        logger.error("Could not open '%s' file", ETC_OS_RELEASE_PATH)
+
+
+def get_rmt_ip_addr():
+    """Return the RMT update server IP the instance is registered to."""
+    try:
+        with open(ETC_HOSTS_PATH, encoding='utf-8') as etc_hosts:
+            etc_hosts_lines = etc_hosts.readlines()
+    except FileNotFoundError:
+        logger.error("Could not open '%s' file", ETC_HOSTS_PATH)
+        return
+
+    for etc_hosts_line in etc_hosts_lines:
+        if 'susecloud.net' in etc_hosts_line:
+            return etc_hosts_line.split('\t')[0]
+
+    if 'cloudregister' in sys.modules:
+        server = get_smt(False)
+        return server.get_ipv4()
+    else:
+        logger.info(
+            'No cloudregister module present, '
+            'can not determine a valid IP address'
```

Comment:
The message here is misleading. AT this point we tried 2 ways to get the IP address, not just using the cloudregister module. AS such it should simply state "Could not determine update server IP address". The first part of the message focuses the reader on the "missing" module.